src/satellitePropagator/physics/YoshidaIntegrator.h

```diff
@@ -0,0 +1,297 @@
+
+    // constants
+    const std::array<double, 2> w {
+        -std::cbrt(2.) / (2. - std::cbrt(2.)),
+        1 / (2. - std::cbrt(2.)),
+    };
+    const std::array<double, 4> c {
+        w[1] / 2.,
+        (w[0] + w[1]) / 2.,
+        (w[0] + w[1]) / 2.,
+        w[1] / 2.,
+    };
+    const std::array<double, 3> d {
+        w[1],
+        w[0],
+        w[1],
+    };
+
+public:
+    /**
+     * @brief Getter function for #delta_t
+     *
+     * @return Value of #delta_t
+     */
+    [[nodiscard]] double getDeltaT() const;
+
+    /**
+     * @brief Setter function for #delta_t
+     *
+     * @param deltaT New value of #delta_t
+     */
+    void setDeltaT(double deltaT);
+
+    /**
+     * @brief Getter function for #accumulator
+     *
+     * @return Value of #accumulator
+     */
+    [[nodiscard]] const Acceleration::AccelerationAccumulator<Container>& getAccumulator() const;
+    Acceleration::AccelerationAccumulator<Container>& getAccumulator();
+
+    /**
+     * @brief Setter function for #accumulator
+     *
+     * @param accumulator New value of #accumulator
+     */
+    void setAccumulator(Acceleration::AccelerationAccumulator<Container>& accumulator);
+
+    /**
+     * @brief Getter function for #container
+     *
+     * @return Value of #container
+     */
+    [[nodiscard]] const Container& getContainer() const;
+    Container& getContainer();
+
+    /**
+     * @brief Setter function for #delta_t
+     *
+     * @param container New value of #delta_t
+     */
+    void setContainer(Container& container);
+};
+
+template <class Container>
+YoshidaIntegrator<Container>::YoshidaIntegrator() = default;
+
+template <class Container>
+YoshidaIntegrator<Container>::~YoshidaIntegrator() = default;
+
+template <class Container>
+void YoshidaIntegrator<Container>::integrate(bool write_time_step) const
+{
+    // do three full and one half sub step
+    for (size_t i = 0; i < 3; ++i) {
+        calculatePosition(c[i]);
+        accumulator->setT(accumulator->getT() + delta_tDiv4);
+        calculateAcceleration(write_time_step);
+        AccelerationUpdate::accelerationUpdate(container);
+        calculateVelocity(d[i]);
+    }
+    calculatePosition(c[3]);
+    accumulator->setT(accumulator->getT() + delta_tDiv4);
+
+    // Le Math: (x_i = ith iteration ; x^j = jth substep
+    // x_i^1 = x_i + c_1 v_i delta t
+    // v_i^1 = v_i + d_1 a(x_i^1) delta t
+    // x_i^2 = x_i^1 + c_2 v_i^1 delta t
+    // v_i^2 = v_i^1 + d_2 a(x_i^2) delta t
+    // x_i^3 = x_i^2 + c_3 v_i^2 delta t
+    // v_i^3 = v_i^2 + d_3 a(x_i^3) delta t
+    // x_i^4 = x_i^3 + c_4 v_i^3 delta t        = x_(i+1)
+    // v_i^4 = v_i^3                            = v_(i+1)
+}
+
+template <class Container>
+void YoshidaIntegrator<Container>::calculatePosition(double c) const
+{
+#ifdef AUTOPAS_OPENMP
+    // autopas works with parallel iterators which it controls itself. No pragma omp for needed!
+#pragma omp parallel
```

Comment:
not sure if still relevant, but you mentioned something you want to pack this all into one big OMP parallel region

Reply:
yes bot not in this PR

Reply:
Will be done in a dedicated PR